packages/utils/src/shorthand.ts

```diff
@@ -0,0 +1,519 @@
+  all: true, // This is a special case, it's a shorthand for all properties
+  animation: [
+    'animation-delay',
+    'animation-direction',
+    'animation-duration',
+    'animation-fill-mode',
+    'animation-iteration-count',
+    'animation-name',
+    'animation-play-state',
+    'animation-timeline',
+    'animation-timing-function',
+  ],
+  'animation-range': ['animation-range-start', 'animation-range-end'],
+  background: [
+    'background-attachment',
+    'background-clip',
+    'background-color',
+    'background-image',
+    'background-origin',
+    'background-position',
+    'background-repeat',
+    'background-size',
+  ],
+  border: [
+    'border-block',
+    'border-block-end',
+    'border-block-end-color',
+    'border-block-end-style',
+    'border-block-end-width',
+    'border-block-start',
+    'border-block-start-color',
+    'border-block-start-style',
+    'border-block-start-width',
+    'border-bottom',
+    'border-bottom-color',
+    'border-bottom-style',
+    'border-bottom-width',
+    'border-inline',
+    'border-inline-end',
+    'border-inline-end-color',
+    'border-inline-end-style',
+    'border-inline-end-width',
+    'border-inline-start',
+    'border-inline-start-color',
+    'border-inline-start-style',
+    'border-inline-start-width',
+    'border-left',
+    'border-left-color',
+    'border-left-style',
+    'border-left-width',
+    'border-right',
+    'border-right-color',
+    'border-right-style',
+    'border-right-width',
+    'border-top',
+    'border-top-color',
+    'border-top-style',
+    'border-top-width',
+  ],
+  'border-block': [
+    'border-block-start',
+    'border-block-start-color',
+    'border-block-start-style',
+    'border-block-start-width',
+    'border-block-end',
+    'border-block-end-color',
+    'border-block-end-style',
+    'border-block-end-width',
+    'border-top-color',
+    'border-top-style',
+    'border-top-width',
+    'border-bottom-color',
+    'border-bottom-style',
+    'border-bottom-width',
+  ],
+  'border-block-end': [
+    'border-block-end-color',
+    'border-block-end-style',
+    'border-block-end-width',
+    'border-top-color',
+    'border-top-style',
+    'border-top-width',
+    'border-bottom-color',
+    'border-bottom-style',
+    'border-bottom-width',
+  ],
+  'border-block-start': [
+    'border-block-start-color',
+    'border-block-start-style',
+    'border-block-start-width',
+    'border-top-color',
+    'border-top-style',
+    'border-top-width',
+    'border-bottom-color',
+    'border-bottom-style',
+    'border-bottom-width',
+  ],
+  'border-bottom': [
+    'border-bottom-color',
+    'border-bottom-style',
+    'border-bottom-width',
+    'border-block-start-color',
+    'border-block-start-style',
+    'border-block-start-width',
+    'border-block-end-color',
+    'border-block-end-style',
+    'border-block-end-width',
+  ],
+  'border-color': [
```

Comment:
Example: Should this sort _above_ `border-top` as well, not just the children?

Reply:
@dddlr I'm not decided enough on what "perfect" looks like for these, so I think it's fine to have some bugs for very rarely used properties…let's circle back to this and call it a gap unless it's obvious.

…in a way, these are "siblings", but to me it feels like there's an explicit order?  MDN doesn't really provide anything and I think there's no explicit browser setting—it's a first-defined-basis:
 - https://developer.mozilla.org/en-US/docs/Web/CSS/border-width
 - https://developer.mozilla.org/en-US/docs/Web/CSS/border-top

But if this makes sense, in theory I think for all of `border-color`, `border-style`, `border-width`, this should be:
```suggestion
  'border-color': [
    'border-top',
    'border-block',
    'border-inline',
    'border-left',
    'border-right',
    'border-bottom',
```

Reply:
ah the ordering of the longhand properties in `shorthandFor` no longer matters - we use `shorthandBuckets` to enforce that `border-color/width/style` comes before `border-top`, in both `packages/css/src/plugins/sort-shorthand-declarations.ts` and `packages/react/src/runtime/sheet.ts`

![Screenshot 2024-09-27 at 10 13 52](https://github.com/user-attachments/assets/392d8086-5c7a-4355-8581-d28b6e0a3444)


i'm happy to reset the order of the longhand properties in `shorthandFor` to alphabetical order to reduce potential confusion

Reply:
ah the order of properties in `shorthandFor` no longer matters - i should probably just reset that to alphabetical order to be honest ...

`shorthandBuckets` is now used in `packages/css/src/plugins/sort-shorthand-declarations.ts` (previously was just in `packages/react/src/runtime/sheet.ts`) to enforce that `border-{color,style,width}` comes before `border-{top,right,bottom,left}`

example https://github.com/atlassian-labs/compiled/pull/1700/files#diff-3b4ae519f265680a755d14f6de289c4d7507a936e4911e95b9b3889fa1ac579bR187

<img width="1707" alt="Screenshot 2024-09-27 at 10 22 54" src="https://github.com/user-attachments/assets/a10e33d0-18d6-4a0f-91e1-f7a6d64dc4a2">

let me know if this aligns with what you have in mind

Reply:
Got it, the test LGTM, that's what this `aShorthandBucket - bShorthandBucket` does 👌